test/inheritAttributes.test.js

```diff
@@ -490,3 +494,869 @@ QUnit.test('end to end - alternate BaseURLs', function(assert) {
   assert.equal(actual.length, 6);
   assert.deepEqual(actual, expected);
 });
+
+QUnit.test(' End to End test for checking support of segments in representation', function(assert) {
+  const actual = inheritAttributes(stringToMpdXml(
+    `
+    <MPD mediaPresentationDuration= "PT30S"  >
+      <BaseURL>https://www.example.com/base/</BaseURL>
+      <Period>
+        <AdaptationSet mimeType= "video/mp4"  >
+          <Role value= "main" ></Role>
+          <SegmentBase indexRangeExact= "true"  indexRange= "820-2087" >
+              <Initialization range= "0-987" />
+          </SegmentBase>
+
+          <Representation
+            mimeType= "video/mp6"
+            bandwidth= "5000000"
+            codecs= "avc1.64001e"
+            height= "404"
+            id= "test"
+            width= "720" >
+            <SegmentBase>
+              <Initialization range= "0-567" />
+            </SegmentBase>
+          </Representation>
+          <Representation
+            height= "545" >
+            <SegmentBase></SegmentBase>
+          </Representation>
+        </AdaptationSet>
+        <AdaptationSet mimeType= "text/vtt"  lang= "en" >
+          <SegmentTemplate></SegmentTemplate>
+          <Representation bandwidth= "256"  id= "en" >
+            <BaseURL>https://example.com/en.vtt</BaseURL>
+          </Representation>
+        </AdaptationSet>
+      </Period>
+    </MPD>
+  `
+  ));
+
+  const expected = [{
+    attributes: {
+      bandwidth: '5000000',
+      baseUrl: 'https://www.example.com/base/',
+      codecs: 'avc1.64001e',
+      height: '404',
+      id: 'test',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'video/mp6',
+      periodIndex: 0,
+      role: {
+        value: 'main'
+      },
+      sourceDuration: 30,
+      width: '720'
+    },
+    segmentInfo: {
+      base: {
+        indexRange: '820-2087',
+        indexRangeExact: 'true',
+        initialization: {
+          range: '0-567'
+        }
+      },
+      list: undefined,
+      template: undefined,
+      timeline: undefined
+    }
+  }, {
+    attributes: {
+      baseUrl: 'https://www.example.com/base/',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'video/mp4',
+      periodIndex: 0,
+      height: '545',
+      role: {
+        value: 'main'
+      },
+      sourceDuration: 30
+    },
+    segmentInfo: {
+      base: {
+        indexRange: '820-2087',
+        indexRangeExact: 'true',
+        initialization: {
+          range: '0-987'
+        }
+      },
+      list: undefined,
+      template: undefined,
+      timeline: undefined
+    }
+  }, {
+    attributes: {
+      bandwidth: '256',
+      baseUrl: 'https://example.com/en.vtt',
+      id: 'en',
+      lang: 'en',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'text/vtt',
+      periodIndex: 0,
+      role: {},
+      sourceDuration: 30
+    },
+    segmentInfo: {
+      base: undefined,
+      list: undefined,
+      template: undefined,
```

Comment:
This should not be undefined, should have used template from containing adaptation set